test/test_cpp_api_parity.py

```diff
@@ -198,18 +220,23 @@ def _compile_cpp_code_inline(self, name, cpp_sources, functions):
         )
         return cpp_module
 
-    # This tests that Python and C++ torch.nn modules have matching constructor arg names and types.
-    def _test_torch_nn_module_ctor_args(self, module_name):
+    def _get_python_module_init_arg_spec(self, module_name):
         python_module_class = getattr(torch.nn, module_name)
-        module_metadata = torch_nn_modules.module_metadata_map[module_name]
-        cpp_default_constructor_args_str = module_metadata.cpp_default_constructor_args
         if PY2:
             init_arg_spec = inspect.getargspec(python_module_class.__init__)
         else:
             init_arg_spec = inspect.getfullargspec(python_module_class.__init__)
+        return init_arg_spec
+
+    # This tests that Python and C++ torch.nn modules have matching constructor arg names and types.
+    def _test_torch_nn_module_ctor_args(self, module_name):
```

Comment:
Nit: This function is getting a little large. After reading it, here is what I think it is doing:
1. check that the # of cpp args is equal to # of python args
2. write code to construct cpp options
3. write code to check existence of all python args in the cpp args.
4. compile the code and run tests

It might be helpful to write comments for (1), (2) and (3), or break them into their own helper functions. However they're still readable (as evidenced by how I am able to figure out what is going on), so it's really up to you. If this function gets larger after this commit then it might be nice to split it up.